File: py/IEC61968/Operations/OutageCauseKind.py
# Converted by an OPENAI API call using model: gpt-3.5-turbo-1106
# outage_cause_kind.py
from enum import Enum


class OutageCauseKind(Enum):
    LIGHTING_STRIKE = 1  # "LIGHTING_STRIKE"
    ANIMAL = 2  # "ANIMAL"
    LINE_DOWN = 3  # "LINE_DOWN"
    POLE_DOWN = 4  # "POLE_DOWN"
    TREE_DOWN = 5  #"TREE_DOWN"

File: java/IEC61968/Operations/OutageCauseKind.java
package IEC61968.Operations;


/**
 * This enumeration describes the primary cause of the outage - planned, unplanned,
 * etc.
 * @author Margaret
 * @version 1.0
 * @created 25-Dec-2023 8:45:23 PM
 */
public enum OutageCauseKind {
	lightingStrike,
	/**
	 * This outage was caused by an animal is was unplanned.  As such it is treated as
	 * a forced outage and is probably classified as "trouble" with a Trouble Ticket
	 * as well as a work/service order.  The primary difference between this and an
	 * unplanned outage is the reason for the outage.  If an animal caused this and
	 * perished as a result, the utility may have other actions that are required to
	 * be taken by the EPA or other groups with whom the utility has an agreement.
	 */
	animal,
	lineDown,
	poleDown,
	treeDown
}